sklearn/naive_bayes.py

```diff
@@ -811,7 +816,11 @@ def _update_feature_log_prob(self, alpha):
         """Apply smoothing to raw counts and compute the weights."""
         comp_count = self.feature_all_ + alpha - self.feature_count_
         logged = np.log(comp_count / comp_count.sum(axis=1, keepdims=True))
-        self.feature_log_prob_ = logged / logged.sum(axis=1, keepdims=True)
+        feature_log_prob = -logged
```

Comment:
Why do we have this negative sign here?

Reply:
The `predict` function in `BaseNB` takes the `argmax`, but `ComplementNB` operates by taking the `argmin`, so the sign needs to be flipped to work correctly (dividing by the sums of rows in `logged`, which are negative values, also accomplishes this goal). But this did make me realize there is a discrepancy in the original paper. The weights are divided by the sum of the absolute value of the weights when the normalization is first described (see equation (8) [here](http://people.csail.mit.edu/jrennie/papers/icml03-nb.pdf)), but the absolute value bars are omitted in the algorithm's pseudocode (Table 4). I'll correct that in the documentation.

Reply:
Thanks for the clarification. Maybe add a small comment in the code, something like:
```py
        # BaseNB.predict uses argmax, but ComplementNB operates with argmin
```